Feature/regional release nceplibs

ywangwof opened the pull request:
I merged aerorahul’s PR #118 for NOAA-EMC/UFS_UTILS to your branch feature/regional_release of UFS_UTILS, which uses project configs from NCEPLIBS for CMake to work. The main changes are _CMakeLists.txt_ files in all _sorc_ directories and a new submodule **CMakeModules** to replace the original "**cmake**" submodule. I tested it on Odin and Stampede only so far. I added two new modulefiles _build.odin_ and _build.stampede_ for the support.

aerorahul committed:
move cmake to CMakeModules

Change-Id: I8e7526ef1b796c3653f06d83d65ccc1d4df2483e

aerorahul committed:
update pointer to CMakeModules

Change-Id: I20a151f91202f78f4b125b6d3b99c8950c265ce7

aerorahul committed:
use imported targets for NCEPLIBS

Change-Id: I332a52f9a1ae79bfa5f05e8565b487f2672f5c7d

aerorahul committed:
update pointer to CMakeModules

aerorahul committed:
update pointer to CMakeModules

aerorahul committed:
update gitmodule to point to correct branch of CMakeModules

aerorahul committed:
removed explicit dependency on png, jasper and zlib as they are resolved in nceplibs. cleaned up nemsio utilities dependency list

aerorahul committed:
ignore exec/ and update build.hera to temporarily point to user test location of nceplibs

aerorahul committed:
CMakeModules was updated. Update pointer to CMakeModules submoule and in .gitmodules

Change-Id: Ia514d413aae62c3440f2d039919d738d443e07df

aerorahul committed:
Merge branch 'develop' into feature/cmakepkgs

aerorahul committed:
update modules on hera orion cray and dell to point to cmake libraries

aerorahul committed:
- utilities linking with nemsio library need to link with bacio_4 and w3nco_d
- There is no Clang or AppleClang Fortran compiler.

aerorahul committed:
remove unused modules crtm, nceppost and g2tmpl

aerorahul committed:
update intel module on cray. Export esmfmkfile variable

aerorahul committed:
update orion paths for netcdf and esmf

aerorahul committed:
update pointer to CMakeModules

aerorahul committed:
use wgrib2 cmake library

Change-Id: I0579ce922e8590b482c9ba3d3c1f50c8fc3c06cf

GeorgeGayno-NOAA committed:
Port repository to Orion

Updates to compile repo on Orion.  

New scripts to run regression tests on Orion.

New script to run the grid creation step on Orion.

Update link_fixdirs.sh to link to Orion fixed directories.

Remove some references to Tide/Gyre.

See #96 and #134 for more details.

aerorahul committed:
update cmakemodues pointer

aerorahul committed:
Merge branch 'develop' into feature/cmakepkgs

aerorahul committed:
update modulefile

aerorahul committed:
fix gfortran issues

ywangwof committed:
Merged with PR #118 to use project configs from NCEPLIBS, tested on Odin and Stampede

ywangwof committed:
Removed version numbers for NCEPLIBS to be future-safe

aerorahul committed:
remove find_package(ip2)

aerorahul committed:
minor cleanup of flags

LarissaReames commented:
@ywangwof can you merge in the latest commits to https://github.com/NOAA-EMC/UFS_UTILS/pull/118 in to your PR just to make sure there aren't any conflicts? Once that's done, I'll merge in your PR.

GeorgeGayno-NOAA committed:
Add lake-maker to grid generation process

Adding the lake maker from Shan Sun and Ning Wang: generate lake fraction and depth on
the FV3 grid and add these records to the orography files. Ensure land fraction and lake 
fraction are consistent.  For details, see issue #111

aerorahul committed:
Merge branch 'develop' into feature/cmakepkgs

Change-Id: I912fe56b6d73909cb55c16c61b6fda9e558e0ffd

kgerheiser committed:
Update modules and CMakeModules (#2)

* Update CMakeModules for latest FindNetCDF that finds MPI for parallel NetCDF and update moduefiles on NOAA machines.

DusanJovic-NOAA committed:
Fix synax error in global_equiv_resol.f90 (using gnu compiler) (#1)

proper fix for GNU syntax errors

kgerheiser committed:
Update Jet and load wgrib2 instead of manually setting wgrib2_ROOT (#3)

kgerheiser committed:
Fix typos and load PNG for Jet

kgerheiser committed:
Load correct wgrib2

Loading Intel after "module use" caused the incorrect version of wgrib2 to be loaded.

Commit:
Merged with #0edc7642 from aerorahul/feature/cmakepkgs

ywangwof committed:
Added machine setup for Stampede

ywangwof commented:
> @ywangwof can you merge in the latest commits to [NOAA-EMC#118](https://github.com/NOAA-EMC/UFS_UTILS/pull/118) in to your PR just to make sure there aren't any conflicts? Once that's done, I'll merge in your PR.

Done and tested building on Odin and Stampede.
